src/algebra/big_operators/finprod.lean

```diff
@@ -700,4 +700,33 @@ lemma finsum_mul {R : Type*} [semiring R] (f : α → R) (r : R)
   (∑ᶠ a : α, f a) * r = ∑ᶠ a : α, f a * r :=
 (add_monoid_hom.mul_right r).map_finsum h
 
+@[to_additive] lemma finset.mul_support_of_fiberwise_prod_subset_image [decidable_eq β]
+  (s : finset α) (f : α → M) (g : α → β) :
+  mul_support (λ b, (s.filter (λ a, g a = b)).prod f) ⊆ s.image g :=
+begin
+  simp only [finset.coe_image, set.mem_image, finset.mem_coe, function.support_subset_iff],
+  intros b h,
+  suffices : (s.filter (λ (a : α), g a = b)).nonempty,
+  { simpa only [s.fiber_nonempty_iff_mem_image g b, finset.mem_image, exists_prop], },
+  exact finset.nonempty_of_prod_ne_one h,
+end
+
+@[to_additive] lemma finprod_mem_finset_of_product (s : finset (α × β)) (f : α × β → M) :
+  ∏ᶠ ab (h : ab ∈ s), f ab = ∏ᶠ a b (h : (a, b) ∈ s), f (a, b) :=
+begin
+  haveI := classical.dec_eq α, haveI := classical.dec_eq β,
+  rw finprod_mem_finset_eq_prod,
+  have : ∀ a, ∏ᶠ b (h : (a, b) ∈ s), f (a, b) = (s.filter (λ ab, prod.fst ab = a)).prod f,
+  { intros a,
+    have ha : ∏ (i : β) in (s.filter (λ ab, prod.fst ab = a)).image prod.snd, f (a, i) =
+      (finset.filter (λ ab, prod.fst ab = a) s).prod f,
+    { apply finset.prod_bij (λ b _, (a, b)); finish, },
+    rw [← ha, ← finprod_mem_finset_eq_prod], simp, },
+  simp_rw this, clear this,
+  rw [finprod_eq_prod_of_mul_support_subset _
+    (s.mul_support_of_fiberwise_prod_subset_image f prod.fst),
+    ← finset.prod_fiberwise_of_maps_to _ f],
+  finish,
+end
```

Comment:
I don't think this is a particularly good lemma, as I can't chain it to handle a product of products. A better lemma IMO might be something like:
```lean
@[to_additive] lemma finprod_curry  (f : α × β → M) (h : (mul_support f).finite) :
  ∏ᶠ ab, f ab = ∏ᶠ a b, f (a, b) :=sorry
```
from which you can recover this lemma as hopefully a tidied version of:
```lean
@[to_additive] lemma finprod_mem_finset_of_product (s : finset (α × β)) (f : α × β → M) :
  ∏ᶠ ab (h : ab ∈ s), f ab = ∏ᶠ a b (h : (a, b) ∈ s), f (a, b) :=
begin
  rw finprod_curry,
  classical,
  have : (λ (ab : α × β), ∏ᶠ (h : ab ∈ s), f ab) = (λ ab, if ab ∈ s then f ab else 1),
  { ext,
    split_ifs,
    rw [eq_true_intro h, finprod_true],  -- there should be a better lemma for this!
    rw [eq_false_intro h, finprod_false], },
  rw this,
  rw mul_support,
  apply (set.finite_mem_finset s).subset,
  intros x hx,
  simp at ⊢ hx,
  exact hx.1,
end
```

Reply:
Hmm, good point and I agree; `finprod_curry` is certainly worth having. I'll put together a proof this afternoon.

I wonder if the following might also be desirable:
```lean
@[to_additive] lemma finprod_mem_finset_of_product' [decidable_eq α] [decidable_eq β]
  (s : finset (α × β)) (f : α × β → M) :
  ∏ᶠ ab (h : ab ∈ s), f ab =
  ∏ᶠ a b (h : b ∈ (s.filter (λ ab, prod.fst ab = a)).image prod.snd), f (a, b) :=
```
For example, given this we can easily prove:
```lean
@[to_additive] lemma finprod_mem_finset_of_product'₃ {γ : Type*}
  (s : finset (α × β × γ)) (f : α × β × γ → M) :
  ∏ᶠ abc (h : abc ∈ s), f abc = ∏ᶠ a b c (h : (a, b, c) ∈ s), f (a, b, c) :=
begin
  classical,
  rw finprod_mem_finset_of_product',
  simp_rw finprod_mem_finset_of_product',
  simp,
end
```

Reply:
I ended up adding various related bits and pieces inspired by your remarks. Not quite sure how much to include so any / all feedback most welcome!